readium/shared/src/main/java/org/readium/r2/shared/OptIn.kt

```diff
@@ -25,6 +25,14 @@ annotation class InternalReadiumApi
 @Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.TYPEALIAS, AnnotationTarget.PROPERTY)
 annotation class ExperimentalReadiumApi
 
+@RequiresOptIn(
+    level = RequiresOptIn.Level.WARNING,
+    message = "This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API."
+)
+@Retention(value = AnnotationRetention.BINARY)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.TYPEALIAS, AnnotationTarget.PROPERTY)
+annotation class DelicateReadiumApi
```

Comment:
Inspired by `@DelicateCoroutinesApi`. We can use it to tag a dangerous Readium API that should be used after carefully reading the doc.